processor/transformer/transformer.go

```diff
@@ -561,3 +580,83 @@ func trackLongRunningTransformation(ctx context.Context, stage string, timeout t
 		}
 	}
 }
+
+func (trans *handle) doFasthttpPost(ctx context.Context, rawJSON []byte, url, stage string, tags stats.Tags) ([]byte, int) {
+	var (
+		retryCount int
+		respData   []byte
+		statusCode int
+		apiVersion []byte
+	)
+
+	err := backoff.RetryNotify(
+		func() error {
+			var reqErr error
+			resp := fasthttp.AcquireResponse()
+			defer fasthttp.ReleaseResponse(resp)
+			requestStartTime := time.Now()
+
+			trace.WithRegion(ctx, "request/post", func() {
+				req := fasthttp.AcquireRequest()
+				defer fasthttp.ReleaseRequest(req)
+				req.SetRequestURI(url)
+				if trans.config.disableKeepAlives {
+					req.Header.SetConnectionClose()
+				}
+				req.Header.SetMethod("POST")
+				req.Header.SetContentType("application/json; charset=utf-8")
+				req.Header.Set("X-Feature-Gzip-Support", "?1")
+				req.Header.Set("X-Feature-Filter-Code", "?1")
+				req.SetBody(rawJSON)
+
+				reqErr = trans.fasthttpClient.Do(req, resp)
+			})
+			trans.requestTime(tags, time.Since(requestStartTime))
+			if reqErr != nil {
+				return reqErr
+			}
+
+			statusCode = resp.StatusCode()
+			if !isJobTerminated(statusCode) && statusCode != StatusCPDown {
+				return fmt.Errorf("transformer returned status code: %v", statusCode)
+			}
+
+			respData = make([]byte, len(resp.Body()))
+			_ = copy(respData, resp.Body())
+
+			apiVersion = make([]byte, len(resp.Header.Peek("apiVersion")))
+			_ = copy(apiVersion, resp.Header.Peek("apiVersion"))
+
+			return nil
+		},
+		backoff.WithMaxRetries(backoff.NewExponentialBackOff(), uint64(trans.config.maxRetry.Load())),
+		func(err error, t time.Duration) {
+			retryCount++
+			trans.logger.Warnf("JS HTTP connection error: URL: %v Error: %+v after %v tries", url, err, retryCount)
+		},
+	)
+	if err != nil {
+		if trans.config.failOnUserTransformTimeout.Load() && stage == userTransformerStage && os.IsTimeout(err) {
+			return []byte(fmt.Sprintf("transformer request timed out: %s", err)), TransformerRequestTimeout
+		} else if trans.config.failOnError.Load() {
+			return []byte(fmt.Sprintf("transformer request failed: %s", err)), TransformerRequestFailure
+		} else {
+			panic(err)
+		}
+	}
+
+	// perform version compatibility check only on success
+	if statusCode == http.StatusOK {
```

Comment:
[optional] Same here, we could use `fasthttp.StatusOK`, although it doesn't matter much, just for consistency.